.Jenkinsfile

```diff
@@ -0,0 +1,47 @@
+pipeline {
+    agent {
+        dockerfile {
+            filename 'Dockerfile'
+            dir 'docker/jenkins'
+            label 'master'
+        }
+    }
+    environment {
+        HOME = '.'
+    }
+    stages {
+        stage('Tests') {
+            options {
+                skipDefaultCheckout()
+            }
+            steps {
+                sh(
+                  label: 'Remove cache in var directory',
+                  script: 'rm -rf var/cache'
+                )
+                withCredentials([string(credentialsId: 'jenkins_github_token', variable: 'GIT_TOKEN')]) {
+                    sh(
+                        label: 'Install/Update sources from Composer',
+                        script: "COMPOSER_AUTH='{\"github-oauth\": {\"github.com\": \"$GIT_TOKEN\"}}\' composer install --no-interaction --no-ansi --no-progress"
+                    )
+                }
+                sh(
+                    label: 'Infection - Running mutation testing',
+                    script: './vendor/bin/infection --threads=1 --min-msi=85 --test-framework-options="--coverage-clover=var/log/phpunit/coverage.xml"'
```

Comment:
infection is very useful, but very costful regarding  time, I would:
- trigger phpunit on PR insread
- cron to trigger infection daily basis, not to have 15+ minutes builds per PR

Reply:
I am discussing it with Serge at the moment how could we improve the performance of the build server. On my macbook (limiting the number of threads to 2) it takes approximately 6-7 minutes to run, I believe we could achieve something similar on Jenkins as well.